# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasource/TrackBuffersManager.cpp
# Commit: cffe107089aa
# Full Hash: cffe107089aa5adcf2191e2edd4bd6b477997cb1
# Author: Paul Adenot <paul@paul.cx>
# Date: 2023-07-07 09:28:00
# Regressor Bug: 1838828
# File Overlap Count: 1
# Description:
#   Bug 1838828 - Downgrade assertions to only crash in Nigthly. r=chunmin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D182914
# ==============================================================================

diff -r 909fc50c54a1 -r cffe107089aa dom/media/mediasource/TrackBuffersManager.cpp
--- a/dom/media/mediasource/TrackBuffersManager.cpp	Fri Jul 07 00:43:34 2023 +0000
+++ b/dom/media/mediasource/TrackBuffersManager.cpp	Fri Jul 07 00:51:27 2023 +0000
@@ -2610,7 +2610,7 @@
       return i;
     }
   }
-  MOZ_CRASH("FindSampleIndex called with invalid arguments");
+  MOZ_ASSERT(false, "FindSampleIndex called with invalid arguments");
 
   return 0;
 }
@@ -2641,8 +2641,8 @@
     // that are less than aFuzz wide, we set a fuzz factor aFuzz/2.
     buffered.SetFuzz(aFuzz / 2);
     TimeIntervals::IndexType index = buffered.Find(aTime);
-    MOZ_RELEASE_ASSERT(index != TimeIntervals::NoIndex,
-                       "We shouldn't be called if aTime isn't buffered");
+    MOZ_ASSERT(index != TimeIntervals::NoIndex,
+               "We shouldn't be called if aTime isn't buffered");
     TimeInterval target = buffered[index];
     target.mFuzz = aFuzz;
     i = FindSampleIndex(track, target);
